internal/profiles/models/models.go

```diff
@@ -15,6 +15,75 @@
 // Package models contains domain models for profiles
 package models
 
+import (
+	"encoding/json"
+	"fmt"
+
+	"github.com/google/uuid"
+
+	"github.com/stacklok/minder/internal/db"
+	minderv1 "github.com/stacklok/minder/pkg/api/protobuf/go/minder/v1"
+)
+
+// ProfileAggregate represents a profile along with rule instances
+type ProfileAggregate struct {
+	ID           uuid.UUID
+	Name         string
+	ActionConfig ActionConfiguration
+	Rules        []RuleInstance
+}
+
+// ActionConfiguration stores the configuration state for a profile
+type ActionConfiguration struct {
+	Remediate ActionOpt
```

Comment:
These fields are isolated into an inner struct because various parts of the engine code previously passed around the whole profile to just get these two values.